pytomorrowio/pytomorrowio.py

```diff
@@ -352,11 +353,8 @@ async def realtime_and_all_forecasts(
         specific fields for specific forecast types, use the corresponding fields list.
         """
         self._num_api_requests = 0
-        if (
-            not all_forecasts_fields
-            and not nowcast_fields
-            and not hourly_fields
-            and not daily_fields
+        if not (
+            all_forecasts_fields or nowcast_fields or hourly_fields or daily_fields
```

Comment:
if `all_forecast_fields` is `[]` then this will return an error even if the other fields are specified. We should definitely remove the `is not None` check below then for consistency

Reply:
I tested my changes pretty thoroughly.
```
>>> [] or ['a'] or ['b'] or None
['a']
```

Reply:
guess i misunderstood how the or worked in this case!